### PR TITLE
Add xhr property to Request

### PR DESCRIPTION
**Pull request by iaddict**
Add up.Request#xhr property which returns the not yet sent XMLHttpRequest object.

With this, it's possible to access the xhr in a [up:request:load](https://unpoly.com/up:request:load) event handler through the event.request.xhr property.

One use case is to add xhr event listeners. For example this is required to listen on upload progress events.

See #435.

### CODE REVIEW

File: src/unpoly/classes/request/xhr_renderer.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ up.Request.XHRRenderer = class XHRRenderer {
   }
 
   buildAndSend(handlers) {
-    this.xhr = new XMLHttpRequest()
+    this.xhr = this.request.xhr
```
Comment (triskweline) [inline src/unpoly/classes/request/xhr_renderer.js:13]: `this.xhr` can just be a local variable now, which compresses better.

Also we no longer need to return the XHR object at the end of the method.

File: src/unpoly/classes/request.js [hunk 2]
```diff
@@ -366,6 +366,18 @@ up.Request = class Request extends up.Record {
     this.badResponseTime ??= u.evalOption(up.network.config.badResponseTime, this)
   }
 
+  /*-
+  Returns the underlying XMLHttpRequest instance.
```
Comment (triskweline) [inline src/unpoly/classes/request.js:370]: Format code reference in backticks:

```
Returns the underlying `XMLHttpRequest` instance.
```

File: src/unpoly/classes/request.js [hunk 3]
```diff
@@ -366,6 +366,18 @@ up.Request = class Request extends up.Record {
...
+
+  @property up.Request#xhr
```
Comment (triskweline) [inline src/unpoly/classes/request.js:372]: To see the type in the docs we must extend this with a `@params` directive:

```
@property up.Request#xhr
@param {XMLHttpRequest} xhr
@stable
```

Sorry for our arcane documentation format!

**Review by triskweline — CHANGES_REQUESTED** [review]
Thank you for the excellent PR!

I suggested a few minor changes in my review.

File: src/unpoly/classes/request/xhr_renderer.js (hunk 1 above)
Comment (iaddict) [reply src/unpoly/classes/request/xhr_renderer.js]: The xhr is now local, which also required to pass xhr to the addHeader method.

**Review by triskweline — APPROVED** [review]

**Comment by triskweline** [comment]
Thank you for your contribution.

**Pull request merged**